### PR TITLE
[VIRTS-2777] PATCH /api/v2/agents/{paw} not updating fields

### PR DESCRIPTION
**Pull request by bleepbop**
## Description

When making a PATCH request to `/api/v2/agents/{paw}`, no fields were updated (the store method in c_agent.py wasn't saving any updated fields). This has been changed to actually store all of the currently updatable fields from the UI.

Note: the `pending_contact` field is listed as an updatable field in the UI, however I don't think it's currently possible to change an agent's contact after creation. I have left `pending_contact` as an updatable field for consistency/in case future behavior changes to allow this.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Verified `PATCH /api/v2/agents/{paw}` has the same functionality as the current UI by using Postman.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by uruwhy** [comment]
Regarding the `pending_contact` and `contact` fields for the agent, the current intended behavior is as follows:
1. If the user wants the agent to switch from contact A to contact B, they will set `pending_contact` to B (currently done via the GUI, but I suppose now it can be done via API). Note that the agent's actual contact has not changed yet - we only marked that we have a pending change available.
2. For future beacons that a contact receives from an agent, the contact will check if the agent's `pending_contact` is different from the current `contact`. If so, it sends a contact change as part of the instructions. If the agent supports contact switches, it'll transition accordingly.

Basically, this makes sure that the agent's `contact` is only updated when an agent sends a beacon via the new contact, since there will be delays between sending the contact switch request and having the agent actually respond. So `pending_contact` is what we want to support changing via API, not `contact` itself.

**Review by argaudreau — APPROVED** [review]
Tested locally and the PATCH endpoint is working as intended.

**Review by uruwhy — CHANGES_REQUESTED** [review]
Could you add a test to `objects/test_agent.py` to confirm this?

**Review by uruwhy — APPROVED** [review]
LGTM
